### PR TITLE
Presentation: Slow native tests' startup

### PR DESCRIPTION
**Pull request by medikornov**
Closes: https://github.com/iTwin/itwinjs-core/issues/5779

Performance improvement:

```
1 test suite - 1 test case
--gtest_filter=RulesDrivenECPresentationManagerContentTests.SelectedNodeInstance_GetNavigationPropertyValue
before: avg 18500 ms
after:  avg  2100 ms
```
   
```
1 test suite - 25 test cases
--gtest_filter=RulesDrivenECPresentationManagerContentTests.SelectedNode*
before: avg 23000 ms
after:  avg  2900 ms
```

```
1 test suite - 151 test cases
--gtest_filter=RulesDrivenECPresentationManagerContentTests.P*
before: avg 24000 ms
after:  avg  6400 ms
```

```
121 test suite - 2530 test cases
before: 584040 ms
after:  392257 ms
```

### CODE REVIEW

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h [hunk 1]
```diff
@@ -34,31 +34,49 @@ BEGIN_ECPRESENTATIONTESTS_NAMESPACE
...
         } \
-    void registry::RegisterSchemaXml(Utf8String name, Utf8String schemaXml) {GetRegisteredSchemaXmls().push_back(bpair<Utf8String, Utf8String>(name, schemaXml));}
+    void registry::RegisterSchemaXml(Utf8String name, Utf8String schemaXml) \
+        { \
+        GetRegisteredSchemaXmls().push_back(bpair<Utf8String, Utf8String>(name, CreateValidSchemaString(name, schema_xml))); \
+        } \
+    void registry::RegisterMultipleSchemasXml(Utf8String name, bvector<Utf8String> schemasXml) \
+        { \
+        int i = 1; \
+        for (auto schema : schemasXml) \
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h:67]:
```suggestion
        for (auto const& schema : schemasXml) \

```

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h [hunk 2]
```diff
@@ -34,31 +34,49 @@ BEGIN_ECPRESENTATIONTESTS_NAMESPACE
 template<typename TRegistry> struct RegisterSchemaHelper
     {
     RegisterSchemaHelper(Utf8String name, Utf8String schemaXml) {TRegistry::RegisterSchemaXml(name, schemaXml);}
+    RegisterSchemaHelper(Utf8String name, bvector<Utf8String> schemasXml) {TRegistry::RegisterMultipleSchemasXml(name, schemasXml);}
+    static Utf8String CreateValidSchemaString(Utf8String name, Utf8String schema_xml)
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h:38]:
```suggestion
    static Utf8String CreateValidSchemaString(Utf8StringCR name, Utf8StringCR schema_xml)

```

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h [hunk 3]
```diff
@@ -34,31 +34,49 @@ BEGIN_ECPRESENTATIONTESTS_NAMESPACE
...
         { \
         static bvector<bpair<Utf8String, Utf8String>> s_registeredSchemaXmls; \
         return s_registeredSchemaXmls; \
         } \
-    void registry::RegisterSchemaXml(Utf8String name, Utf8String schemaXml) {GetRegisteredSchemaXmls().push_back(bpair<Utf8String, Utf8String>(name, schemaXml));}
+    void registry::RegisterSchemaXml(Utf8String name, Utf8String schemaXml) \
+        { \
+        GetRegisteredSchemaXmls().push_back(bpair<Utf8String, Utf8String>(name, CreateValidSchemaString(name, schema_xml))); \
+        } \
+    void registry::RegisterMultipleSchemasXml(Utf8String name, bvector<Utf8String> schemasXml) \
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h:64]:
```suggestion
    void registry::RegisterMultipleSchemasXml(Utf8StringCR name, bvector<Utf8String> const& schemasXml) \

```

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h [hunk 4]
```diff
@@ -34,31 +34,49 @@ BEGIN_ECPRESENTATIONTESTS_NAMESPACE
 template<typename TRegistry> struct RegisterSchemaHelper
     {
     RegisterSchemaHelper(Utf8String name, Utf8String schemaXml) {TRegistry::RegisterSchemaXml(name, schemaXml);}
+    RegisterSchemaHelper(Utf8String name, bvector<Utf8String> schemasXml) {TRegistry::RegisterMultipleSchemasXml(name, schemasXml);}
```
Comment (grigasp) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.h:37]:
```suggestion
    RegisterSchemaHelper(Utf8StringCR name, bvector<Utf8String> const& schemasXml) {TRegistry::RegisterMultipleSchemasXml(name, schemasXml);}

```

**Review by grigasp — CHANGES_REQUESTED** [review]

**Comment by grigasp** [comment]
I don't understand why the performance of running all the tests improved, do you?
```
121 test suite - 2530 test cases
before: 584040 ms
after:  392257 ms
```

**Comment by medikornov** [comment]
> Maybe, when setting up schemas for test suite X, we were importing schemas of ALL tests, including from other test suites?

Yes, that's the reason.

Every test suite setup runs this function `InitSchemaRegistry`. As an argument in this function we get all collected schemas from all tests. 
- Previously: we imported all these schemas into `db` for each test suite. This means that even if a test suite has only 1 test case with 1 required schema, all 900+ schemas were imported into db.
- Now: we import only the schemas needed for that test suite. So if a test suite has only 1 test case with 1 required schema, we only import that 1 required schema. The others are not imported.

**Review by grigasp — APPROVED** [review]

File: iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp [hunk 5]
```diff
@@ -24,6 +36,8 @@ void RulesEngineTestHelpers::InitSchemaRegistry(ECDbR ecdb, bvector<bpair<Utf8St
             BeAssert(false);
             continue;
             }
+        if (testNamesThatShouldRun.find(pair.first) == testNamesThatShouldRun.end())
```
Comment (saskliutas) [inline iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp:39]: Shouldn't this check be done before reading schema xml?

Comment (medikornov) [reply iModelCore/ECPresentation/Tests/NonPublished/Helpers/TestHelpers.cpp]: Fixed

**Pull request merged**